src/arch/amdgpu/vega/insts/instructions.hh

```diff
@@ -43915,78 +43916,232 @@ namespace VegaISA
+              const int B, typename T1, typename T2>
+    class Inst_VOP3P_MAI__V_MFMA : public Inst_VOP3P_MAI {
+
+    private:
+      static constexpr int gprs_a = M * K * B / 64, gprs_b = K * N * B / 64,
+                           gprs_c_d = M * N * B / 64;
+
+    public:
+      Inst_VOP3P_MAI__V_MFMA(InFmt_VOP3P_MAI *iFmt)
+          : Inst_VOP3P_MAI(iFmt, (_delta == 2)
+                                     ? "v_mfma_f64_" + std::to_string(M) + "x" +
+                                           std::to_string(N) + "x" +
+                                           std::to_string(K) + "f64"
+                                     : "v_mfma_f32_" + std::to_string(M) + "x" +
+                                           std::to_string(N) + "x" +
+                                           std::to_string(K) + "f32") {
+        setFlag(ALU);
+      }
+      ~Inst_VOP3P_MAI__V_MFMA() {}
+
+      int getNumOperands() override {
+        return numDstRegOperands() + numSrcRegOperands();
+      } // getNumOperands
+
+      int numDstRegOperands() override { return 1; }
+      int numSrcRegOperands() override { return 3; }
+
+      int getOperandSize(int opIdx) override {
+        switch (opIdx) {
+        case 0: // src0 "A"
+          return 4*gprs_a;
+        case 1: // src1 "B"
+          return 4*gprs_b;
+        case 2: // src2 "C"
+          return 4*gprs_c_d;
+        case 3: // dst
+          return 4*gprs_c_d;
+        default:
+          fatal("op idx %i out of bounds\n", opIdx);
+          return -1;
+        }
+      } // getOperandSize
 
-    class Inst_VOP3P_MAI__V_MFMA_I32_16X16X16I8 : public Inst_VOP3P_MAI
+    void
+    execute(GPUDynInstPtr gpuDynInst) override
     {
-      public:
-        Inst_VOP3P_MAI__V_MFMA_I32_16X16X16I8(InFmt_VOP3P_MAI*);
-        ~Inst_VOP3P_MAI__V_MFMA_I32_16X16X16I8();
 
-        int
-        getNumOperands() override
-        {
-            return numDstRegOperands() + numSrcRegOperands();
-        } // getNumOperands
+        int acc_cd_off = 0;
+        int acc_a_off = 0;
+        int acc_b_off = 0;
+        if (instData.ACC_CD) {
+            acc_cd_off = gpuDynInst->wavefront()->accumOffset;
+        }
+        if (extData.ACC) {
+            int tmp_acc = extData.ACC;
+            if (tmp_acc & 0x1) {
+                acc_a_off = gpuDynInst->wavefront()->accumOffset;
+            }
+            if (tmp_acc & 0x2) {
+                acc_b_off = gpuDynInst->wavefront()->accumOffset;
+            }
+        }
 
-        int numDstRegOperands() override { return 1; }
-        int numSrcRegOperands() override { return 3; }
+        typename std::aligned_storage<sizeof(T1), alignof(T1)>::type _src0[gprs_a];
+        typename std::aligned_storage<sizeof(T1), alignof(T1)>::type _src1[gprs_b];
+        typename std::aligned_storage<sizeof(T1), alignof(T1)>::type _src2[gprs_c_d];
+        typename std::aligned_storage<sizeof(T2), alignof(T2)>::type _vdst[gprs_c_d];
```

Comment:
I've recently discovered this is going to be deprecated in C++23: https://stackoverflow.com/questions/71828288/why-is-stdaligned-storage-to-be-deprecated-in-c23-and-what-to-use-instead

I've already made a change to fix this myself.